xarray/core/variable.py

```diff
@@ -1361,8 +1361,11 @@ def reduce(self, func, dim=None, axis=None,
 
         if dim is not None:
             axis = self.get_axis_num(dim)
-        data = func(self.data if allow_lazy else self.values,
-                    axis=axis, **kwargs)
+        if axis is not None:
+            data = func(self.data if allow_lazy else self.values,
+                        axis=axis, **kwargs)
+        else:
+            data = func(self.data if allow_lazy else self.values, **kwargs)
```

Comment:
This no longer passes `axis=None` into `func`, which I think is reasonable - it's generally the default. (but someone should confirm my thinking)

Reply:
This looks good to me, though I would rewrite this to an intermediate variable for `self.data if allow_lazy else self.values` rather than writing it out twice

Reply:
Done, should be ready for merging now.